tensorflow_datasets/image/B_All_White_Blood_Cancer_Microscopic_Image.py

```diff
@@ -0,0 +1,117 @@
+
+_CITATION = """\
+@article{gupta2017stain,
+  title={Stain Color Normalization and Segmentation of Plasma Cells in Microsc
+  opic Images as a Prelude to Development of Computer Assisted Automated Disea
+  se Diagnostic Tool in Multiple Myeloma},
+  author={Gupta, Ritu and Mallick, Pramit and Duggal, Rahul and Gupta, Anubha
+  and Sharma, Ojaswa},
+  journal={Clinical Lymphoma, Myeloma and Leukemia},
+  volume={17},
+  number={1},
+  pages={e99},
+  year={2017},
+  publisher={Elsevier}
+}
+@inproceedings{duggal2016overlapping,
+  title={Overlapping cell nuclei segmentation in microscopic images using deep
+  belief networks},
+  author={Duggal, Rahul and Gupta, Anubha and Gupta, Ritu and Wadhwa, Manya
+  and Ahuja, Chirag},
+  booktitle={Proceedings of the Tenth Indian Conference on Computer Vision,
+  Graphics and Image Processing},
+  pages={82},
+  year={2016},
+  organization={ACM}
+}
+@article{duggal2016segmentation,
+  title={Segmentation of overlapping/touching white blood cell nuclei using
+  artificial neural networks},
+  author={Duggal, R and Gupta, A and Gupta, R},
+  journal={CME Series on Hemato-Oncopathology, All India Institute of Medical
+  Sciences (AIIMS). New Delhi, India},
+  year={2016}
+}
+@inproceedings{duggal2017sd,
+  title={SD-layer: stain deconvolutional layer for CNNs in medical microscopic
+  imaging},
+  author={Duggal, Rahul and Gupta, Anubha and Gupta, Ritu and Mallick, Pramit}
+  ,
+  booktitle={International Conference on Medical Image Computing and Computer-
+  Assisted Intervention},
+  pages={435--443},
+  year={2017},
+  organization={Springer}
+}
+//@under review{
+title={GCTI-SN: Geometry-Inspired Chemical and Tissue Invariant Stain Normaliz
+ation of Microscopic Medical Images},
+author={Anubha Gupta, Rahul Duggal, Ritu Gupta, Lalit Kumar, Nisarg Thakkar,
+and Devprakash Satpathy}
+}
+
+"""
+
+_DESCRIPTION = """\
+ A large data set of  B-ALL white blood cancer microscopic images. The dataset
+ contains total 1867 cropped images, which 1219 malignant cells images are
+ labeled as class 1, 647 normal cells images are labeled as class 0. The size
+ of each image is roughly 300 x 300 x 3 RGB.
+"""
+
+
+class BAllWhiteBloodCancerMicroscopicImage(tfds.core.GeneratorBasedBuilder):
+    """This is a two class classification problem."""
+    VERSION = tfds.core.Version('0.1.0')
+
+    def _info(self):
+        return tfds.core.DatasetInfo(
+            builder=self,
+            description=_DESCRIPTION,
+            features=tfds.features.FeaturesDict({
+                "image": tfds.features.Image(),
+                "label": tfds.features.ClassLabel(names=["cancer", "normal"]),
+            }),
+            supervised_keys=("image", "label"),
+            urls=["https://competitions.codalab.org/competitions/20395"],
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        extracted_path = dl_manager.manual_dir
```

Comment:
These collections are freely available to browse, download, and use for commercial, scientific and educational purposes as outlined in the Creative Commons Attribution 3.0 Unported License. Questions may be directed to help@cancerimagingarchive.net. Please be sure to acknowledge both this data set and TCIA in publications by including the following citations in your work:

You must have an automatic download process. No manual download.